src/core/reference/include/openvino/reference/bitwise_and.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+
+#include "openvino/reference/autobroadcast_binop.hpp"
+
+namespace ov {
+namespace reference {
+namespace func {
+// Check for char datatype used by ov::element::boolean
+template <class T, typename std::enable_if<std::is_same<typename std::decay<T>::type, char>::value>::type* = nullptr>
+T bitwise_and(const T arg0, const T arg1) {
+    return static_cast<T>(std::bit_and<bool>()(arg0, arg1));
```

Comment:
On each iteration the functor is created then do calculation and destroyed. Functor should be created once before loop then only its call operator is used.

This should be corrected to create functor before iterations.